runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -506,31 +542,46 @@ public void onGameTick(GameTick event)
 		Player player = client.getLocalPlayer();
 		WorldPoint currentWorldPoint = player.getWorldLocation();
 
-		if (freezeTimer == null)
+		if (freezeTimer != null)
+		{
+			// assume movement means unfrozen
+			if (!currentWorldPoint.equals(lastPoint))
+			{
+				removeGameTimer(freezeTimer.getTimer());
+				freezeTimer = null;
+			}
+		}
+
+		lastPoint = currentWorldPoint;
+
+		if (!widgetHiddenChangedOnPvpWorld)
 		{
-			lastPoint = currentWorldPoint;
 			return;
 		}
 
-		// assume movement means unfrozen
-		if (!currentWorldPoint.equals(lastPoint))
+		widgetHiddenChangedOnPvpWorld = false;
+
+		Widget widget = client.getWidget(PVP_WORLD_SAFE_ZONE);
+		if (widget != null
+			&& !widget.isSelfHidden())
 		{
-			removeGameTimer(freezeTimer.getTimer());
-			freezeTimer = null;
-			lastPoint = currentWorldPoint;
+			log.debug("Entered safe zone in PVP world, clearing Teleblock timer.");
+			removeTbTimers();
 		}
 	}
 
 	@Subscribe
 	public void onGameStateChanged(GameStateChanged gameStateChanged)
 	{
-		// Check for game state changes which removes a teleblock (Hopping or Login screen)
-		if (gameStateChanged.getGameState() == GameState.HOPPING || gameStateChanged.getGameState() == GameState.LOGIN_SCREEN)
+		switch (gameStateChanged.getGameState())
 		{
-			removeGameTimer(HALFTB);
-			removeGameTimer(DMM_HALFTB);
-			removeGameTimer(FULLTB);
-			removeGameTimer(DMM_FULLTB);
+			case HOPPING:
+			case LOGIN_SCREEN:
+				removeTbTimers();
+				break;
+			case LOGGED_IN:
+				loggedInRace = true;
```

Comment:
When do you actually set this variable to false?

Reply:
I think this got lost during some rebasing or rework... This was meant to be set to false on game tick, so I've re-added that.